src/main/java/com/checkmarx/flow/sastscanning/ScanRequestConverter.java

```diff
@@ -48,15 +48,23 @@ public CxScanParams toScanParams(ScanRequest scanRequest) throws CheckmarxExcept
 
     private void setScanConfiguration(ScanRequest scanRequest, Integer projectId) {
         if (entityExists(projectId)) {
-            log.debug("Scan request will contain scan configuration of the existing project.");
-            CxScanSettings scanSettings = scannerClient.getScanSettingsDto(projectId);
-            if (scanSettings != null && scanSettings.getEngineConfigurationId() != null) {
-                Integer configId = scanSettings.getEngineConfigurationId();
-                String configName = scannerClient.getScanConfigurationName(configId);
-                log.debug("Using scan configuration ID: {}, name: '{}'.", configId, configName);
-                scanRequest.setScanConfiguration(configName);
-            } else {
-                log.warn("Unable to retrieve scan settings for the existing project (ID {}).", projectId);
+
+            if (!ScanUtils.empty(scanRequest.getScanConfiguration())){
+                String scanConfiguration = scanRequest.getScanConfiguration();
+                log.debug("overriding scan configuration with '{}'", scanConfiguration);
+                scanRequest.setScanConfiguration(scanConfiguration);
+            }
```

Comment:
Something won't make sense to me - if scan request does has the scan configuration value, why setting it again inside scan request?

Reply:
Fixed